examples/rxJavaExample/src/main/java/io/realm/examples/rxjava/gotchas/GotchasActivity.java

```diff
@@ -52,151 +50,97 @@
  * - https://github.com/realm/realm-java/issues/1208
  * - https://github.com/realm/realm-java/issues/931
  */
-public class GotchasActivity extends Activity {
+public class GotchasActivity extends AppCompatActivity {
     private Realm realm;
-    private Subscription subscription;
+    private Disposable disposables;
     private ViewGroup container;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_gotchas);
-        container = (ViewGroup) findViewById(R.id.list);
+        container = findViewById(R.id.list);
         realm = Realm.getDefaultInstance();
     }
 
     @Override
     protected void onResume() {
         super.onResume();
 
-        Subscription distinctSubscription = testDistinct();
-        Subscription bufferSubscription = testBuffer();
-        Subscription subscribeOnSubscription = testSubscribeOn();
+        Disposable distinctDisposable = testDistinct();
+        Disposable bufferDisposable = testBuffer();
+        Disposable subscribeOnDisposable = testSubscribeOn();
 
         // Trigger updates
-        realm.executeTransaction(new Realm.Transaction() {
-            @Override
-            public void execute(Realm realm) {
-                realm.where(Person.class).findAllSorted( "name", Sort.ASCENDING).get(0).setAge(new Random().nextInt(100));
-            }
-        });
-
-        subscription = new CompositeSubscription(
-                distinctSubscription,
-                bufferSubscription,
-                subscribeOnSubscription
-        );
+        realm.executeTransaction(r ->
+                r.where(Person.class).findAllSorted( "name", Sort.ASCENDING).get(0).setAge(new Random().nextInt(100)));
+
+        disposables = new CompositeDisposable(distinctDisposable, bufferDisposable, subscribeOnDisposable);
     }
 
     /**
      * Shows how to be careful with `subscribeOn()`
      */
-    private Subscription testSubscribeOn() {
-        Subscription subscribeOn = realm.asObservable()
-                .map(new Func1<Realm, Person>() {
-                    @Override
-                    public Person call(Realm realm) {
-                        return realm.where(Person.class).findAllSorted("name").get(0);
-                    }
-                })
+    private Disposable testSubscribeOn() {
+        Disposable subscribeOn = realm.asFlowable()
+                .map(realm -> realm.where(Person.class).findAllSorted("name").get(0))
```

Comment:
Not sure why this was added?

Reply:
Well it was already there :smile:

Reply:
I'll remove that map operator and there are still weird things in example such as a useless button and other subtle things.

Reply:
The intent of `testSubscribeOn` seems to show that  accessing Realm was created on the UI thread on `Schedulers.io()` will crash. So I think I would not change this.